Automattic/wp-calypso
# Hosting onboarding flow: use new plans grid

Related to https://github.com/Automattic/dotcom-forge/issues/2126.

## Proposed Changes

This PR brings back the new plan grid on hosting after a [revert PR](https://github.com/Automattic/wp-calypso/pull/75628) was merged.

The fix was to forward the prop from the plans step container and also I fixed the conditional. It was "working" on the hosting flow because the conditional was flipped, resulting in buggy behavior.

## Testing Instructions

1. Checkout this branch;
2. Verify that `/start/plans` include the Enterprise plan;
3. Verify that `/start/hosting/plans-hosting` does not include the Enterprise plan.

Of course, make sure to have the `onboarding/2023-pricing-grid` feature flag turned on.
